docs/sorting-algorithms/insertion-sort.mdx

```diff
@@ -0,0 +1,194 @@
+---
+title: Insertion Sort
+slug: /sorting-algorithm/insertion-sort
+---
+
+## Introduction
+
+A simple sorting algorithm which works on the idea of picking elements from the unsorted part one by one and inserting them into their right positions in the sorted part. The array is virtually divided into 2 parts, a sorted one and an unsorted one. Just like we arrange a group of shuffled pages according to their page numbers, values form unsorted part are picked and placed at the right positions in the sorted portion.
+
+![Imgur](https://i.imgur.com/Ora9Q1J.png)
+
+## Algorithm
+
+For sorting an array of size n in ascending order:
+1. Start your loop from arr[1] till arr[n].
+2. Compare the selected element of the current iteration (key) to the element before it.
+3. If this key is smaller than its predecessor, compare it to the elements before the predecessor.
+4. Shift all the greater elements up by one position to make space for the smaller element.
+
+## Code
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+  ]
+}>
+
+<TabItem value="cpp">
+
+```cpp
+// EXECUTION OF INSERTION SORT IN C++
+
+/*First we  will  compare first element in the list with its adjacent element.
+And at every comparison if the element can be inserted at a particular position, 
+then it is inserted by shifting the other elements one position to the right.
+The above steps are repeated until all the elements in the list are in their 
+appropriate position.i.e. list is in sorted order.*/
+
+
+
+
+#include <iostream>
+using namespace std;
+
+//Function to print array.
+void display(int arr[], int size) 
+{ 
+    int i; 
+    for (i=0; i < size; i++) 
+    cout<< arr[i]<<"\t"; 
+    cout<<"\n"; 
+} 
+
+//Main function to run the program.
+int main() 
+{ 
+    int array[] = {5, 3, 1, 9, 8, 2, 4,7}; 
```

Comment:
Can you make it dynamic input? I understand that you used the code from NeoAlgo, and since we had a laxly maintenance in the past, we are fixing that up. You can change it up here and also send a PR to the NeoAlgo, if you want, to be compensated for the efforts.